### DISCUSSION TITLE
How to setup redirect for all emails which have no account or alias ? (email to *@example.com redirected to a catch all mailbox ?)

### DISCUSSION THREAD

**Discussion by shodanx2** [docker-mailserver/docker-mailserver#2873] [Q&A] [answered]
I'm not sure what this feature is properly called.
But my current email at google, I can receive email sent to anything@example.com  without having to create a mailbox or alias first. 

How can I setup something equivalent with docker-mailserver ?

thanks !

**Comment by casperklein** [comment, marked as answer]
https://docker-mailserver.github.io/docker-mailserver/edge/faq/#how-can-i-configure-a-catch-all

**Comment by shodanx2** [comment]
thank you !

**Comment by shodanx2** [comment]
Oops ! 
I have implemented this solution however there is a problem

suppose I have email accounts
user1@example.com
user2@example.com
and alias 
AliasToUser1@example.com


when I added to docker-data/dms/config/postfix-virtual.cf
@example.com user1@example.com

Now there is a problem that ALL mail goes to user1@example.com

Is there a way to have only the remainder go to the catch all address ?

example

NotAUser@example.com   would go to user1@example.com  without having to create this alias

and the other already existing addresses would go to their respective mailbox ?

**Reply by casperklein** [reply]
See here: https://github.com/docker-mailserver/docker-mailserver/issues/1454

**Comment by shodanx2** [comment]
Thank you !

So create an alias from each user account to each user account did the trick.
I suspect that when an emails comes in, it tries to match an alias first and match user account second. 


so the solution in terms of commands was 

./setup.sh alias add user1@example.com user1@example.com
./setup.sh alias add user2@example.com user2@example.com
./setup.sh alias add @example.com user1@example.com

**Reply by casperklein** [reply]
Great to hear. If you want, you can enhance our documentation regarding this point 👍

**Comment by shodanx2** [comment]
I suggest the following text for https://docker-mailserver.github.io/docker-mailserver/edge/faq/#how-can-i-configure-a-catch-all

------------------------------------------------------------------------------------------------------------------------

This will redirect all incoming mail for example.com to user1@example.com

`./setup email add @example.com user1@example.com`

NOTE : Currently, this redirect will also affect email destined to existing mailboxes

To prevent this, also create an alias to each mailbox

`./setup alias add user2@example.com user2@example.com`

**Reply by casperklein** [reply]
On the top right of the mentioned page, there is an edit button to easily do a change and open a pull-request. Then we can review your change 👍

Let me know, if that's okay for you.
